### PR TITLE
hyprland: 0.33.1 -> 0.34.0

### PR DESCRIPTION
**Pull request by MikaelFangel**
## Description of changes

Updates hyprland to: [0.34.0](https://github.com/hyprwm/Hyprland/releases/tag/v0.34.0)
[Changelog](https://github.com/hyprwm/Hyprland/compare/v0.33.1...v0.34.0)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by MikaelFangel** [comment]
Result of `nixpkgs-review pr 278347` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>grimblast</li>
    <li>hdrop</li>
    <li>hyprland</li>
    <li>hyprland.dev</li>
    <li>hyprland.man</li>
    <li>hyprlandPlugins.hy3</li>
    <li>hyprshot</li>
    <li>waybar</li>
    <li>xdg-desktop-portal-hyprland</li>
  </ul>
</details>

**Review by fufexan — COMMENTED** [review]
Between `0.33.1` and `0.34.0`, the `date` field was added to `hyprctl version`.
https://github.com/hyprwm/Hyprland/compare/v0.33.1...v0.34.0#diff-4e3bd328e4c998f53b19da7639df45b2db63601be4b4343534880adfbeaa5840

Not sure if we can extract the date from the `src`, so we may want to hardcode it.

**Comment by MikaelFangel** [comment]
@fufexan I've tried to add the date, but hardcoded it. (:

File: pkgs/applications/window-managers/hyprwm/hyprland/default.nix [hunk 1]
```diff
@@ -66,6 +67,7 @@ stdenv.mkDerivation (finalAttrs: {
       --replace "@HASH@" '${finalAttrs.src.rev}' \
       --replace "@BRANCH@" "" \
       --replace "@MESSAGE@" "" \
+      --replace "@DATE@" "20231206" \
```
Comment (fufexan) [inline pkgs/applications/window-managers/hyprwm/hyprland/default.nix:70]: My bad, I was not explicit enough. In the flake we use ISO 8601 formatting, so in my case the output looks like `Date: 2023-12-27`.

I think it's more readable this way :)

Comment (MikaelFangel) [reply pkgs/applications/window-managers/hyprwm/hyprland/default.nix]: Done! Thanks for your time (:

**Review by fufexan — APPROVED** [review]
LGTM!

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**